meteor/client/ui/Settings/Studio/Generic.tsx

```diff
@@ -347,3 +250,34 @@ export const StudioGenericProperties = withTranslation()(
 		}
 	}
 )
+
+function NewShowStyleButton() {
+	const history = useHistory()
+
+	const onShowStyleAdd = useCallback(() => {
+		MeteorCall.showstyles
+			.insertShowStyleBase()
+			.then((showStyleBaseId) => {
+				history.push('/settings/showStyleBase/' + showStyleBaseId)
+			})
+			.catch(console.error)
+	}, [])
```

Comment:
I don't think the `useCallback` is necessary here. You could just wrap the entire thing into a `React.memo` and since it's not taking any props, it will just render once. Same for `RedirectToShowStyleBtutton`.